bin/oref0-pump-loop.sh

```diff
@@ -1,82 +1,116 @@
 #!/bin/bash
 
-# main pump-loop
-main() {
+# old pump-loop
+old_main() {
     prep
-    until( \
-        echo && echo Starting pump-loop at $(date): \
-        && wait_for_bg \
-        && wait_for_silence \
-        && if_mdt_get_bg \
-        && refresh_old_pumphistory_enact \
-        && refresh_old_pumphistory_24h \
-        && refresh_old_profile \
-        && touch monitor/pump_loop_enacted -r monitor/glucose.json \
-        && refresh_temp_and_enact \
-        && refresh_pumphistory_and_enact \
-        && refresh_profile \
-        && refresh_pumphistory_24h \
-        && echo Completed pump-loop at $(date) \
-        && touch monitor/pump_loop_completed -r monitor/pump_loop_enacted \
-        && echo); do
-
-            if grep -q "percent" monitor/temp_basal.json; then
-                echo "Pssst! Your pump is set to % basal type. The pump won’t accept temporary basal rates in this mode. Change it to absolute u/hr, and temporary basal rates will then be able to be set."
-            fi
-            # On a random subset of failures, mmtune
-            echo Error, retrying \
-            && maybe_mmtune
-            sleep 5
-    done
+    if ! overtemp; then
+        until( \
```

Comment:
doesn't this `until` prevent a graceful shutdown of the pump loop with `/etc/init.d/cron stop`. 
In 0.5.0 the pump loop can be stopped by stopping cron and it will finish after the pump loop (and once the other loops are finished).

Reply:
Yes, and that's why we don't do that in main().  This is legacy code that's no longer called (but still useful for people trying to get 512 pumps to work, etc.)